### PR TITLE
Competition sellers json crawler 

### PR DESCRIPTION
**Pull request by sonaisrayel**
[Task](https://dev.azure.com/onlinemediasolutions/HelpDesk/_workitems/edit/3312)
    
   Sellers/Competitors crawler will work every day with cron job
  Add api to create and get competitors
  Add validation for competitors url
  Email send functionality

### CODE REVIEW

File: workers/sellers/helpers.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+func (worker *Worker) prepareEmail(competitorsData []CompetitorData, err error) error {
+	if len(competitorsData) > 0 {
+		now := time.Now()
+		today := now.Format("2006-01-02")
+		yesterday := now.AddDate(0, 0, -1).Format("2006-01-02")
+
+		subject := fmt.Sprintf("Competitors sellers.json daily changes - %s", today)
+		message := fmt.Sprintf("Below are the sellers.json changes between - %s and %s", yesterday, today)
+
+		err = SendCustomHTMLEmail("sonai@onlinemediasolutions.com", "sonai@onlinemediasolutions.com", subject, message, competitorsData)
```
Comment (finkiel) [inline workers/sellers/helpers.go:198]: you should change the email before deploying

File: workers/sellers/helpers.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+		jobs <- competitor
+	}
+
+	close(jobs)
+	wg.Wait()
+	close(results)
+	return results
+}
+
+func (worker *Worker) prepareEmail(competitorsData []CompetitorData, err error) error {
```
Comment (finkiel) [inline workers/sellers/helpers.go:189]: the output of the email looks great - very clear

File: workers/sellers/helpers.go [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+	if len(competitorsData) > 0 {
+		now := time.Now()
+		today := now.Format("2006-01-02")
```
Comment (finkiel) [inline workers/sellers/helpers.go:192]: the data can be saved as const (you are using the same one in 2 places and it's always the same

File: utils/email.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	mailer.SetHeader("Bcc", emailReq.Bcc)
+	mailer.SetHeader("Subject", emailReq.Subject)
+
+	if emailReq.IsHTML {
+		mailer.SetBody("text/html", emailReq.Body)
+	} else {
+		mailer.SetBody("text/plain", emailReq.Body)
+	}
+
+	dialer := gomail.NewDialer("smtp.gmail.com", 465, "smtp@onlinemediasolutions.com", "sqmrvlxfljsjkyhh")
```
[the remark this answers is not in the document]
Comment (sonaisrayel) [reply utils/email.go]: About email I asked Gadi for help

**Review by finkiel — APPROVED** [review]

**Review by finkiel — APPROVED** [review]

**Pull request merged**